breakpad-symbols/src/http.rs

```diff
@@ -375,12 +416,12 @@ pub fn unpack_cabinet_file(
     let mut reader = cab.read_file(&cab_file)?;
 
     // Now try to create the temp cache file (not yet in the cache)
-    let mut temp = create_cache_file(tmp, &final_cache_path)?;
-    std::io::copy(&mut reader, &mut temp)?;
-
+    let mut temp = create_cache_file(&final_cache_path).await?;
+    let mut std_file = temp.file.into_std().await;
+    std::io::copy(&mut reader, &mut std_file)?;
+    temp.file = fs::File::from_std(std_file);
     // And swap it into the cache
-    temp.persist_noclobber(&final_cache_path)
-        .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
+    fs::rename(&temp.temp_path, &temp.dest_path).await?;
```

Comment:
This code sucks and isn't properly async because `cab` returns its own Reader that isn't async. It's in a disabled-by-default-even-in-minidump-stackwalk path though, so it doesn't affect my current numbers.